misc/bazel/pkg.bzl

```diff
@@ -106,8 +107,11 @@ def _extract_pkg_filegroup_impl(ctx):
         dest_src_map = {}
         for dest, file in pfi.dest_src_map.items():
             file_kind, expanded_dest = _expand_path(dest, platform)
+            if file_kind == "generic" and dest in arch_overrides:
+                file_kind = "arch"
```

Comment:
another mechanism could be to use another marker like `{CODEQL_ARCH_SPECIFIC}` that would expand to nothing but would mark the files as arch-specific. That would limit the changes required in just `_expand_path`, without any new attribute.

Reply:
I thought about that, but imo that's
* very magic - why would `CODEQL_ARCH_SPECIFIC` expand to nothing, while `CODEQL_PLATFORM` expands to a string?
* We'd need to put the file we need this for into a separate filegroup just to set the correct prefix.
So I think a separate attribute is more explicit for the reader, and easier to look up the documentation for.